According to [1], braces in regexps should now be escaped, otherwise a deprecation warning is
displayed.

[1] http://perldoc.perl.org/perldelta.html#A-literal-%22{%22-should-now-be-escaped-in-a-pattern

Index: trunk/scripts/services/amavis
===================================================================
--- trunk/scripts/services/amavis	(revision 293)
+++ trunk/scripts/services/amavis	(working copy)
@@ -2283,7 +2283,7 @@
       #XXX elsif (($action, $key, $ip, $from, $to) = ( $p1 =~ /^(?:Virus found - quarantined|(?:(Passed|Blocked) )?INFECTED) \(([^\)]+)\),[A-Z .]*(?: \[($re_IP)\])?(?: \[$re_IP\])* [<(]([^>)]*)[>)] -> [(<]([^(<]+)[(>]/o ))
 
       # the first IP is the envelope sender.
-      if ($p1 !~ /^(CLEAN|SPAM(?:MY)?|INFECTED \(.*?\)|BANNED \(.*?\)|BAD-HEADER(?:-\d)?|UNCHECKED|MTA-BLOCKED|OVERSIZED|OTHER|TEMPFAIL)(?: {[^}]+})?, ([^[]+ )?(?:([^<]+) )?[<(](.*?)[>)] -> ([(<].*?[)>]), (?:.*Hits: ([-+.\d]+))(?:.* size: (\d+))?(?:.* autolearn=(\w+))?/) {
+      if ($p1 !~ /^(CLEAN|SPAM(?:MY)?|INFECTED \(.*?\)|BANNED \(.*?\)|BAD-HEADER(?:-\d)?|UNCHECKED|MTA-BLOCKED|OVERSIZED|OTHER|TEMPFAIL)(?: \{[^}]+})?, ([^[]+ )?(?:([^<]+) )?[<(](.*?)[>)] -> ([(<].*?[)>]), (?:.*Hits: ([-+.\d]+))(?:.* size: (\d+))?(?:.* autolearn=(\w+))?/) {
          inc_unmatched('passblock');
          next;
       }
Index: trunk/scripts/services/postfix
===================================================================
--- trunk/scripts/services/postfix	(revision 293)
+++ trunk/scripts/services/postfix	(working copy)
@@ -1847,7 +1847,7 @@
         # KeyboardInterrupt 
         $line =~ /^Read line: "/ or
         $line =~ /^Found the end of entry$/ or
-        $line =~ /^Config: {/ or
+        $line =~ /^Config: \{/ or
         $line =~ /^spfcheck: pyspf result/ or
         $line =~ /^Starting$/ or
         $line =~ /^Normal exit$/ or
Index: trunk/scripts/services/secure
===================================================================
--- trunk/scripts/services/secure	(revision 293)
+++ trunk/scripts/services/secure	(working copy)
@@ -506,7 +506,7 @@
       $DeniedAccess{"$User,$Reason"}++;
    } elsif ($ThisLine =~ /^request-key: Cannot find command to construct key/) {
       $RequestKeyFailures++;
-   } elsif (my ($type,$from,$response,$client,$service,$e) = ($ThisLine =~ /krb5kdc\[[0-9]*\]: (AS_REQ|TGS_REQ) \([0-9]+ etypes {[ 0-9]+}\) ([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+): (ISSUE|UNKNOWN_SERVER): authtime [0-9]+, (?:etypes {rep=[0-9]+ tkt=[0-9]+ ses=[0-9]+},)? ([^ ]+) for ([^ ,]+)(?:, )?(.*)$/)) {
+   } elsif (my ($type,$from,$response,$client,$service,$e) = ($ThisLine =~ /krb5kdc\[[0-9]*\]: (AS_REQ|TGS_REQ) \([0-9]+ etypes \{[ 0-9]+}\) ([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+): (ISSUE|UNKNOWN_SERVER): authtime [0-9]+, (?:etypes \{rep=[0-9]+ tkt=[0-9]+ ses=[0-9]+},)? ([^ ]+) for ([^ ,]+)(?:, )?(.*)$/)) {
      if($service=~/^krbtgt\/([^@]+)@\1/) {
         $service='Login';
      }
@@ -515,7 +515,7 @@
         $e='';
      }
      $KerbList{$response}{$type}{$from}{$service}{$client}{$e}++;
-  } elsif (my ($type,$from,$response,$client,$service,$e) = ($ThisLine =~ /krb5kdc\[[0-9]*\]: (AS_REQ|TGS_REQ) \([0-9]+ etypes {[ 0-9]+}\) ([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+): (NEEDED_PREAUTH|PREAUTH_FAILED|CLIENT_NOT_FOUND): ([^ ]+) for ([^ ,]+)(?:, )?(.*)$/)) {
+  } elsif (my ($type,$from,$response,$client,$service,$e) = ($ThisLine =~ /krb5kdc\[[0-9]*\]: (AS_REQ|TGS_REQ) \([0-9]+ etypes \{[ 0-9]+}\) ([0-9]+\.[0-9]+\.[0-9]+\.[0-9]+): (NEEDED_PREAUTH|PREAUTH_FAILED|CLIENT_NOT_FOUND): ([^ ]+) for ([^ ,]+)(?:, )?(.*)$/)) {
      if($service=~/^krbtgt\/([^@]+)@\1/) {
         $service='Login';
      }
Index: trunk/scripts/services/http
===================================================================
--- trunk/scripts/services/http	(revision 293)
+++ trunk/scripts/services/http	(working copy)
@@ -310,13 +310,13 @@
    } elsif ($logformat =~ /\G%O/gc) {
       $parse_field[$parse_index][$parse_subindex++] = "bytes_out";
       $parse_string[$parse_index] .= "(-|\\d*)";
-   } elsif ($logformat =~ /\G%{Referer}i/gci) {
+   } elsif ($logformat =~ /\G%\{Referer}i/gci) {
       $parse_string[$parse_index] .= "(.*)";
       $parse_field[$parse_index][$parse_subindex++] = "referrer";
-   } elsif ($logformat =~ /\G%{User-Agent}i/gci) {
+   } elsif ($logformat =~ /\G%\{User-Agent}i/gci) {
       $parse_string[$parse_index] .= "(.*)";
       $parse_field[$parse_index][$parse_subindex++] = "agent";
-   } elsif ($logformat =~ /\G%({.*?})?./gc) {
+   } elsif ($logformat =~ /\G%(\{.*?})?./gc) {
       $parse_string[$parse_index] .= "(.*?)";
       $parse_field[$parse_index][$parse_subindex++] = "not_used";
    } elsif ($logformat =~ /\G\|/gc) {
